cabal-install/Distribution/Client/Utils/Parsec.hs

```diff
@@ -0,0 +1,65 @@
+module Distribution.Client.Utils.Parsec (
+    renderParseError,
+    ) where
+
+import Distribution.Client.Compat.Prelude
+import Prelude ()
+
+import qualified Data.ByteString       as BS
+import qualified Data.ByteString.Char8 as BS8
+
+import Distribution.Parsec.Common
+       (PError (..), PWarning (..), Position (..), showPError, showPWarning, zeroPos)
+import Distribution.Simple.Utils  (fromUTF8BS)
+
+-- | Render parse error highlighting the part of the input file.
```

Comment:
Doc nit:

```
-- | Render parse errors and warnings, highlighting the part of the input file.
```

Reply:
Feel free to improve further. I merged this as is.